pandas/core/groupby/grouper.py

```diff
@@ -312,7 +311,9 @@ def _get_grouper(
         return grouper, obj
 
     @final
-    def _set_grouper(self, obj: NDFrame, sort: bool = False) -> None:
+    def _set_grouper(
+        self, obj: NDFrame, sort: bool = False, *, gpr_index: Index | None = None
+    ):
```

Comment:
Can the result be typed as NDFrame, Index?

Reply:
looks like adding that causes a cascade of other complaints.  id like to handle those separately